app/src/main/java/com/example/unsplashphotos/domain/usecase/PhotoUseCase.kt

```diff
@@ -0,0 +1,24 @@
+package com.example.unsplashphotos.domain.usecase
+
+
+import androidx.paging.Pager
+import androidx.paging.PagingConfig
+import androidx.paging.PagingData
+import androidx.paging.PagingSource
+import com.example.unsplashphotos.data.model.Photo
```

Comment:
Domain layer UseCases shouldn't have data layer's Photo object. Use different objects for different layers. One Photo class with all the annotations required for JSON parsing and database and keep it in data layer. Another Photo class containing only the required fields for using in domain and/or presentation layer and keep it in domain layer. Use a mapper for converting to/from these different objects.
See this article for details on how to separate these two types of objects.
https://proandroiddev.com/the-real-repository-pattern-in-android-efba8662b754

Reply:
Additionally, you can extract an interface from the UseCase that doesn't have any dependency on the data layer. These interfaces will reside on the domain layer. In the data layer, the implementations of these UsesCases will reside, containing dependencies from the data layer (i.e. Retrofit interface or repository). ViewModels will use the domain layer's UseCases and thus the implementation will be abstracted out.